server/game/cards/locations/04/salladhorsaan.js

```diff
@@ -0,0 +1,34 @@
+const DrawCard = require('../../../drawcard.js');
+
+class SalladhorSaan extends DrawCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                afterChallenge: (event, challenge) => (
+                    challenge.winner === this.controller &&
+                    challenge.isParticipating(this))
+            },
+            handler: () => {
+                this.game.promptForSelect(this.controller, {
+                    activePromptTitle: 'Select a card to put into play',
+                    waitingPromptTitle: 'Waiting for opponent to use ' + this.name,
+                    cardCondition: card => (
+                        card.location === 'hand' &&
+                        card.controller === this.controller &&
+                        ((card.hasTrait('Warship') && card.getType() === 'location') ||
+                        (card.hasTrait('Weapon') && card.getType() === 'attachment'))),
+                    onSelect: (player, card) => {
+                        player.moveCard(card, 'play area');
```

Comment:
This will work for the locations but not for attachments. Attachments require you choose something to attach the card on, which is done from `Player.promptForAttachment`.